mobile/cmp/multiZoneGrid/impl/MultiZoneMapperModel.ts

```diff
@@ -0,0 +1,363 @@
+ */
+import '@xh/hoist/mobile/register';
+import {HoistModel, managed, XH} from '@xh/hoist/core';
+import {span} from '@xh/hoist/cmp/layout';
+import {action, bindable, computed, makeObservable, observable} from '@xh/hoist/mobx';
+import {StoreRecord} from '@xh/hoist/data';
+import {Column, ColumnRenderer, GridModel} from '@xh/hoist/cmp/grid';
+import {checkbox} from '@xh/hoist/mobile/cmp/input';
+import {wait} from '@xh/hoist/promise';
+import {cloneDeep, isBoolean, isEmpty, isEqual, isString} from 'lodash';
+import {ReactNode} from 'react';
+import {MultiZoneGridModel} from '../MultiZoneGridModel';
+import {Zone, ZoneLimit, ZoneMapping} from '../Types';
+
+export interface MultiZoneMapperConfig {
+    /** The MultiZoneGridModel to be configured. */
+    multiZoneGridModel: MultiZoneGridModel;
+
+    /** True (default) to show Reset button to restore default configuration. */
+    showRestoreDefaults?: boolean;
+
+    /** True (default) to group columns by their chooserGroup */
+    groupColumns?: boolean;
+}
+
+/**
+ * State management for the MultiZoneMapper component.
+ *
+ * It is not necessary to manually create instances of this class within an application.
+ * @internal
+ */
+export class MultiZoneMapperModel extends HoistModel {
+    multiZoneGridModel: MultiZoneGridModel;
+    showRestoreDefaults: boolean;
+    groupColumns: boolean;
+
+    @managed
+    gridModel: GridModel;
+
+    @observable
+    isOpen: boolean = false;
+
+    @bindable
+    selectedZone: Zone = 'tl';
+
+    @observable.ref
+    mappings: Record<Zone, ZoneMapping[]>;
+
+    fields: MapperField[] = [];
+    sampleRecord: StoreRecord;
+
+    @computed
+    get isDirty(): boolean {
+        return !isEqual(this.mappings, this.multiZoneGridModel.mappings);
+    }
+
+    get leftFlex(): number {
+        const ret = this.multiZoneGridModel.leftColumnSpec?.flex;
+        return isBoolean(ret) ? 1 : ret ?? 2;
+    }
+
+    get rightFlex(): number {
+        const ret = this.multiZoneGridModel.rightColumnSpec?.flex;
+        return isBoolean(ret) ? 1 : ret ?? 1;
+    }
+
+    get limits(): Partial<Record<Zone, ZoneLimit>> {
+        return this.multiZoneGridModel.limits;
+    }
+
+    get delimiter(): string {
+        return this.multiZoneGridModel.delimiter;
+    }
+
+    constructor(config: MultiZoneMapperConfig) {
+        super();
+        makeObservable(this);
+
+        const {multiZoneGridModel, showRestoreDefaults = true, groupColumns = true} = config;
+
+        this.multiZoneGridModel = multiZoneGridModel;
+        this.showRestoreDefaults = showRestoreDefaults;
+        this.groupColumns = groupColumns;
+
+        this.fields = this.getFields();
+        this.gridModel = this.createGridModel();
+
+        this.addReaction(
+            {
+                track: () => this.selectedZone,
+                run: () => this.syncMapperDataForZone()
+            },
+            {
+                track: () => XH.routerState,
+                run: this.close
```

Comment:
nit: Line 100 and Line 96 should both be either dynamically bound, or both be fat arrow functions